driver-core/src/main/java/com/datastax/driver/core/ArrayBackedRow.java

```diff
@@ -239,6 +242,23 @@ public InetAddress getInet(String name) {
         return getInet(metadata.getFirstIdx(name));
     }
 
+    public Token getToken(int i) {
+        if (tokenFactory == null)
+            throw new DriverInternalError("Token factory not set. This should only happen at initialization time");
+
+        metadata.checkType(i, tokenFactory.getTokenType().getName());
+
+        ByteBuffer value = data.get(i);
+        if (value == null || value.remaining() == 0)
+            return null;
+
+        return tokenFactory.deserialize(value);
+    }
+
+    public Token getToken(String name) {
+        return getToken(metadata.getFirstIdx(name));
```

Comment:
I don't really see the problem: what `getToken(String name)` should do is first check if it can find `"token(" + name + ")"`, and if it can't, it checks name directly for the case where we've used an alias. If we have both "k" and "token(k)" in the metadata and you call `getToken("k")`, there is virtually no change you meant to fetch and try to deserialze "k" as a token, especially since in most case it won't work anyway.

Additionally, we could first check if `name.startsWith("token(")` and avoid trying to prepend in that case, which is what I called the "smarter" version, but that part if optional imo.